app/models/theme_field.rb

```diff
@@ -568,6 +568,12 @@ def invalidate_baked!
     if (will_save_change_to_value? || will_save_change_to_upload_id?) && !will_save_change_to_value_baked?
       self.value_baked = nil
     end
+    if will_save_change_to_upload_id?
```

Comment:
We want this to apply to existing theme_fields which were created before this commit. How about removing this conditional? `javascript_cache.content=` is super cheap if it exists already.

Reply:
Its tricky ... cause we would download an upload from the CDN to do this, it can get costly. 

I hear you, during the transition this may get annoying - will make some minor adjustment to protect us during upgrade

- If extension is .js
- AND either upload changed OR javascript_cache is missing
- THEN do the download / copy